src/hipscat_import/map_reduce.py

```diff
@@ -49,14 +37,17 @@ def map_to_pixels(
 
     # Load file using appropriate mechanism
     if "csv" in file_format:
-        data = dd.read_csv(input_file)
+        if schema_file:
+            schema_parquet = pd.read_parquet(schema_file)
+            data = pd.read_csv(input_file, dtype=schema_parquet.dtypes.to_dict())
+        else:
+            data = pd.read_csv(input_file)
 
     elif file_format == "fits":
-        data = dd.from_pandas(
-            Table.read(input_file, format="fits").to_pandas(), chunksize=500_000
-        )
+        data = Table.read(input_file, format="fits").to_pandas()
     elif file_format == "parquet":
-        data = dd.read_parquet(input_file, engine="pyarrow")
+        data = pd.read_parquet(input_file, engine="pyarrow")
+        data.reset_index(inplace=True)
     else:
         raise NotImplementedError(f"File Format: {file_format} not supported")
```

Comment:
If we want to add more formats in the future, this could probably be refactored (e.g. using a factory instead of all the if statements), but it's fine for now